Content.Server/Chemistry/ReagentEffectConditions/InCrit.cs

```diff
@@ -0,0 +1,25 @@
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.MobState;
+using Content.Shared.MobState.Components;
+using Content.Shared.MobState.EntitySystems;
+
+namespace Content.Server.Chemistry.ReagentEffectConditions
+{
+    public sealed class InCrit : ReagentEffectCondition
+    {
+
+        [Dependency] private readonly SharedMobStateSystem _mobState = default!;
+
+        public override bool Condition(ReagentEffectArgs args)
+        {
+            if (args.EntityManager.TryGetComponent(args.SolutionEntity, out MobStateComponent? mobState))
+            {
+                if (mobState.CurrentState == DamageState.Critical)
```

Comment:
generalize this with a datafield so that you can pick any damagestate. you wouldn't want to have to duplicate this effect for alive/critical/dead.

Reply:
Changed it into DamageStateCondition, this should do